collections/nemo_nlp/nemo_nlp/transformer/utils.py

```diff
@@ -35,7 +35,7 @@ def form_attention_mask(input_mask, diagonal=None):
         future_mask = torch.tril(
             torch.ones(attn_shape).byte().to(input_mask.device), diagonal)
         attn_mask = attn_mask & future_mask
-    attention_mask = (1 - attn_mask.to(input_mask.dtype)) * NEG_INF
+    attention_mask = (1 - attn_mask.to(torch.float)) * NEG_INF
```

Comment:
1. what's the motivation?
2. does it work with fp16?

Reply:
1. "If input_mask is an int type, transformer training would fail silently."
2. @AlexGrinch said the attention softmax is done in float32 anyways. And yes my training uses this and using O1

Reply:
yes, softmax is done in fp32